frontend/viewer/src/ProjectView.svelte

```diff
@@ -317,100 +326,92 @@
-              $selectedEntry = $selectedEntry;
+            on:change={(e) => {
+              updateEntryInList($selectedEntry = e.detail.entry);
               $entries = $entries;
             }}
             on:delete={onEntryDeleted} />
         {:else}
           <div class="w-full h-full z-10 bg-surface-100 flex flex-col gap-4 grow items-center justify-center text-2xl opacity-75">
             No entry selected
             {#if !readonly}
-              <NewEntryDialog on:created={e => onEntryCreated(e.detail.entry)}/>
+              <NewEntryButton on:click={() => openNewEntryDialog()} />
             {/if}
           </div>
         {/if}
       </div>
-      <div class="side-scroller pl-6 border-l-2 gap-4 flex flex-col col-start-3" class:border-l-2={$selectedEntry && !expandList} class:sm-view:border-l-2={pickedEntry && !readonly} class:sm-view:hidden={!pickedEntry || readonly} class:lg-view:hidden={expandList}>
+      <div class="side-scroller pl-6 border-l-2 gap-4 flex-col col-start-3 hidden" class:border-l-2={$selectedEntry && !expandList} class:lg-view:flex={!expandList}>
         {#if $selectedEntry}
           <div class="sm-form:hidden" class:sm:hidden={expandList}>
-            <Button icon={collapseActionBar ? mdiArrowExpandLeft : mdiArrowCollapseRight} class="text-field-sibling-button" iconOnly rounded variant="outline" on:click={() => collapseActionBar = !collapseActionBar} />
+            <Button icon={$state.rightToolbarCollapsed ? mdiArrowExpandLeft : mdiArrowCollapseRight} class="text-field-sibling-button" iconOnly rounded variant="outline"
+              on:click={() => $state.rightToolbarCollapsed = !$state.rightToolbarCollapsed} />
           </div>
         {/if}
-        <div class="sm-form:w-auto w-[15vw] collapsible-col max-sm:self-center" class:self-center={collapseActionBar} class:lg-view:collapse-col={expandList} class:!w-min={collapseActionBar}>
+        <div class="w-[15vw] collapsible-col sm-form:w-min" class:self-center={$state.rightToolbarCollapsed} class:lg-view:collapse-col={expandList} class:!w-min={$state.rightToolbarCollapsed}>
           {#if $selectedEntry}
             <div class="contents" class:lg-view:hidden={expandList}>
               <div class="h-full flex flex-col gap-4 justify-stretch">
-                {#if !readonly}
-                  <div class="contents" bind:this={entryActionsElem}>
-
-                  </div>
-                {/if}
-                {#if $selectedEntry}
-                  <div class="contents">
-<!--                    button must be a link otherwise it won't follow the redirect to a protocol handler-->
-                    <OpenInFieldWorksButton entryId={$selectedEntry.id} {projectName} show={$features.openWithFlex}/>
-                  </div>
-                {/if}
-                <div class="contents sm-form:hidden" class:hidden={collapseActionBar}>
+                <div class="grid gap-4 auto-rows-fr">
+                  {#if !readonly}
```

Comment:
why don't we have the portal in readonly view? shouldn't the buttons inside determine if they should show based on readonly?

Reply:
You're quite right 👍. This was broken. The history button wasn't available in readonly mode.
In the process, I decided it probably makes sense to have the Configure button available when in the editor (on mobile)